apps/passport-server/src/database/queries/telegram/fetchTelegramEvent.ts

```diff
@@ -154,3 +160,60 @@ export async function fetchUserTelegramChats(
   );
   return result.rows[0] ?? null;
 }
+
+export type ChatIDWithEventsAndMembership = ChatIDWithEventIDs & {
+  isChatMember: boolean;
+};
+
+// Fetch a list of Telegram chats that can be joined with the status of user
+// The list is sorted such that chat a user hasn't joined are returned first
+export async function fetchTelegramChatsWithMembershipStatus(
+  client: Pool,
+  userId: number
+): Promise<ChatIDWithEventsAndMembership[]> {
+  const result = await sqlQuery(
+    client,
+    `
+    SELECT
+      tbe.telegram_chat_id AS "telegramChatID",
+        ARRAY_AGG(tbe.ticket_event_id) AS "ticketEventIds",
+        CASE WHEN tbc.telegram_user_id IS NOT NULL THEN true ELSE false END AS "isChatMember"
+    FROM 
+        telegram_bot_events tbe 
+    LEFT JOIN 
+        telegram_bot_conversations tbc 
+    ON 
+        tbe.telegram_chat_id = tbc.telegram_chat_id AND tbc.telegram_user_id = $1
+    GROUP BY 
+        tbe.telegram_chat_id, tbc.telegram_user_id
+    ORDER BY 
+        "isChatMember" ASC;
+    `,
+    [userId]
+  );
+  return result.rows;
+}
+
+export async function fetchTelegramTopic(
+  client: Pool,
+  telegramChatId: number,
+  topicId: number,
+  onlyAnon?: boolean
```

Comment:
nit: checking if `is_anon_topic` is `true` in the returned `TelegramTopic` seems cleaner to me, but this is fine too. Up to you

Reply:
I agree, updating